cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1545,26 +1553,114 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
             List<Future> futures = new ArrayList<>(pvcs.size());
 
             for (PersistentVolumeClaim pvc : pvcs)  {
-                futures.add(pvcOperations.reconcile(namespace, pvc.getMetadata().getName(), pvc));
+                Future<Void> result = Future.future();
+
+                pvcOperations.getAsync(namespace, pvc.getMetadata().getName()).setHandler(res -> {
+                    if (res.succeeded())    {
+                        PersistentVolumeClaim realPvc = res.result();
+
+                        if (realPvc == null || realPvc.getStatus() == null || !"Bound".equals(realPvc.getStatus().getPhase())) {
+                            // This branch handles the following conditions:
+                            // * The PVC doesn't exist yet, we should create it
+                            // * The PVC is not Bound and we should reconcile it
+                            pvcOperations.reconcile(namespace, pvc.getMetadata().getName(), pvc).setHandler(pvcRes -> {
+                                if (pvcRes.succeeded()) {
+                                    result.complete();
+                                } else {
+                                    result.fail(pvcRes.cause());
+                                }
+                            });
+                        } else if (realPvc.getStatus().getConditions().stream().filter(cond -> "Resizing".equals(cond.getType()) && "true".equals(cond.getStatus().toLowerCase(Locale.ENGLISH))).findFirst().orElse(null) != null)  {
+                            // The PVC is Bound but it is already resizing => Nothing to do, we should let it resize
+                            log.debug("{}: The PVC {} is resizing, nothing to do", reconciliation, pvc.getMetadata().getName());
+                            result.complete();
+                        } else if (realPvc.getStatus().getConditions().stream().filter(cond -> "FileSystemResizePending".equals(cond.getType()) && "true".equals(cond.getStatus().toLowerCase(Locale.ENGLISH))).findFirst().orElse(null) != null)  {
+                            // The PVC is Bound and resized but waiting for FS resizing => We need to restart the pod which is using it
+                            String podName = cluster.getPodName(getPodIndexFromPvcName(pvc.getMetadata().getName()));
+                            fsResizingRestartRequest.add(podName);
+                            log.info("{}: The PVC {} is waiting for file system resizing and the pod {} needs to be restarted.", reconciliation, pvc.getMetadata().getName(), podName);
+                            result.complete();
+                        } else {
+                            // The PVC is Bound and resizing is not in progress => We should check if the SC supports resizing and check if size changed
+                            Quantity currentSize = realPvc.getStatus().getCapacity().get("storage");
```

Comment:
Consider factoring the contents of this block into a separate method, just because this method is rather large.

Reply:
Done, I split it into several methods to make them a bit smaller.